src/paperless_mail/oauth.py

```diff
@@ -0,0 +1,127 @@
+import asyncio
+import logging
+from datetime import timedelta
+
+from django.conf import settings
+from django.utils import timezone
+from httpx_oauth.oauth2 import OAuth2
+from httpx_oauth.oauth2 import OAuth2Token
+from httpx_oauth.oauth2 import RefreshTokenError
+
+from paperless_mail.models import MailAccount
+
+
+class PaperlessMailOAuth2Manager:
+    GMAIL_OAUTH_ENDPOINT_TOKEN = "https://accounts.google.com/o/oauth2/token"
+    GMAIL_OAUTH_ENDPOINT_AUTH = "https://accounts.google.com/o/oauth2/auth"
+    OUTLOOK_OAUTH_ENDPOINT_TOKEN = (
+        "https://login.microsoftonline.com/common/oauth2/v2.0/token"
+    )
+    OUTLOOK_OAUTH_ENDPOINT_AUTH = (
+        "https://login.microsoftonline.com/common/oauth2/v2.0/authorize"
+    )
+
+    def __init__(self):
+        self._gmail_client = None
+        self._outlook_client = None
+
+    @property
+    def gmail_client(self) -> OAuth2:
+        if self._gmail_client is None:
+            self._gmail_client = OAuth2(
```

Comment:
I know, still not helpful, but was there a reason to not use [GoogleOAuth2 ](https://frankie567.github.io/httpx-oauth/reference/httpx_oauth.clients/#httpx_oauth.clients.google.GoogleOAuth2) or [MicrosoftGraphOAuth2](https://frankie567.github.io/httpx-oauth/reference/httpx_oauth.clients/#httpx_oauth.clients.microsoft.MicrosoftGraphOAuth2)?

I promise I'll actually try this soon (tm)

Reply:
Haha, no rush. I tried them at first and they didnt work, but I also fixed some other stuff after that which may have been the cause. I can circle back and try them again.